File: leetcode3/남효정/1539. Kth Missing Positive Number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        miss_num = 0
        cnt = 0
        while True:
            miss_num += 1
            if miss_num not in arr:
                cnt += 1
            if k == cnt:
                return miss_num

File: leetcode3/염혜정/1539. Kth Missing Positive Number.java
// O(n)...

class Solution {
    public int findKthPositive(int[] arr, int k) {
        Set<Integer> set = new HashSet<>();
        for (int a : arr) set.add(a);
        
        int i = 1;
        int cnt = 0;
        int result = 0;
        while(true) {
            if (!set.contains(i)) cnt++;
            if (cnt == k) {
                result = i;
                break;
            }
            i++;
        }
        return result;
    }
}
